p2p/peerManager/peerdb/datastore_test.go

```diff
@@ -116,3 +111,94 @@ func TestQuery(t *testing.T) {
 		require.Equal(t, key, "/"+peers[0].AddrInfo.ID.String())
 	})
 }
+
+func TestGetSize(t *testing.T) {
+	ps, teardown := setupDB(t)
+	defer teardown()
+
+	peers := createPeers(t, 5)
+
+	keys := make([]datastore.Key, len(peers))
+
+	cases := []struct {
+		i       int
+		Entropy uint64
+		PubKey  []byte
+	}{
+		{
+			i:       0,
+			Entropy: uint64(12345),
+			PubKey:  []byte(""),
+		},
+		{
+			i:       1,
+			Entropy: uint64(1234567890),
+			PubKey:  []byte("pub"),
+		},
+		{
+			i:       2,
+			Entropy: uint64(1),
+			PubKey:  []byte("pubkey"),
+		},
+		{
+			i:       3,
+			Entropy: uint64(0),
+			PubKey:  []byte("pubkey1234567890"),
+		},
+		{
+			i:       4,
+			Entropy: uint64(12345678901234),
+			PubKey:  []byte("pubkey12345678901234567890"),
+		},
+	}
+
+	var wg sync.WaitGroup
+
+	// Add value to keys and test first time
+	for i, peer := range peers {
+		wg.Add(1)
+		go func(peer *PeerInfo, i int) {
+			defer wg.Done()
+			keys[i] = datastore.NewKey(peer.AddrInfo.ID.String())
+			value, err := json.Marshal(peer)
+			require.NoError(t, err)
+
+			err = ps.Put(context.Background(), keys[i], value)
+			require.NoError(t, err)
+			size, err := ps.GetSize(context.Background(), keys[i])
+			require.NoError(t, err)
+			require.Equal(t, len(value), size)
+		}(peer, i)
+	}
+	wg.Wait()
+
+	// Update keys is parallel and check if size is updated
```

Comment:
Great that's better, thanks. However the more important thing for the bug I'm chasing down is actually the size when rapidly adding and removing elements. Can you make a copy of this test and modify it to add and remove many keys rapidly? We don't need to check the value so you can reuse that.